app/controllers/application_controller.rb

```diff
@@ -44,6 +45,15 @@ def create_user_event(event_type, user = current_user)
     Event.create(user_id: user.id, event_type: event_type)
   end
 
+  def find_user_events(event_type, user = current_user)
```

Comment:
this method is pretty specific to one use case, it shouldn't be on the controller that all other controllers inherit from. ditto for `mail_spammed?`

We can move these into a service object, or a helper